src/lib/utils.rs

```diff
@@ -187,6 +198,187 @@ To compress an uncompressed FASTQ file with bgzip:
     Err(anyhow!(message).context(context))
 }
 
+pub static INPUT_FASTQ_SUFFIX: &str = "_001.fastq.gz";
+
+lazy_static! {
+    /// <*>_L00#_<R# or I#>_001.fastq.gz
+    static ref INPUT_FASTQ_REGEX: Regex = Regex::new(r"^(.*)_L00(\d{1})_([RIUS])(\d{1})_001.fastq.gz$").unwrap();
+}
+
+/// Contains information about a FASTQ that has been inferred from the file name.  This includes
+/// lane, segment type (e.g. template/read, sample barcode/index, molecular barcode/index, and
+/// skip bases), and segment number (e.g. R1 or R2).
+#[derive(Debug, Clone)]
+pub struct InputFastq {
+    pub path: PathBuf,
+    pub prefix: String,
+    pub lane: usize,
+    pub kind: SegmentType,
+    pub kind_number: i32,
+}
+
+impl InputFastq {
+    /// Create a new `InputFastq` inferring information from the file name.  This must match the
+    /// `INPUT_FASTQ_REGEX` pattern.
+    pub fn new<P: AsRef<Path>>(path: P) -> Option<InputFastq> {
+        let file_name = path.as_ref().file_name().unwrap().to_string_lossy();
+        match INPUT_FASTQ_REGEX.captures(&file_name) {
+            None => None,
+            Some(captures) => {
+                let prefix = captures.get(1).unwrap().as_str().to_string();
+                let lane = captures.get(2).unwrap().as_str().parse::<usize>().unwrap();
```

Comment:
unwraps, unwraps everywhere!
Maybe add a comment as to why these unwrap()s are safe?